src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -6093,7 +6097,15 @@ private void MoveDirectoryInfoUnchecked(DirectoryInfo directory, string destinat
             // is just a question of the drive
             if (IsSameWindowsVolume(directory.FullName, destinationPath))
             {
-                directory.MoveTo(destinationPath);
+                try
+                {
+                    directory.MoveTo(destinationPath);
+                }
+                catch (IOException e) when (e.HResult == ERROR_ACCESS_DENIED)
+                {
+                    // If move doesn't work because the source and destination are on different volumes (like DFS), fall back to CopyAndDelete
+                    CopyAndDelete(directory, destinationPath, force);
+                }
             }
             else
             {
```

Comment:
Since we have a fallback to CopyAndDelete() on the exception it makes no sense to call IsSameWindowsVolume(). We could simplify the code,  and remove the check at all:
```c#
if (IsSameWindowsVolume(directory.FullName, destinationPath))
...
else
...
```

Reply:
That's true, will update